### PR TITLE
Run a check for ack floor drift.

### PR DESCRIPTION
**Pull request by derekcollison**
Also periodically check. If all normal will be very cheap.

Signed-off-by: Derek Collison <derek@nats.io>

### CODE REVIEW

File: server/consumer.go [hunk 1]
```diff
@@ -3237,13 +3237,82 @@ func (o *consumer) hbTimer() (time.Duration, *time.Timer) {
...
+			o.processTerm(seq, p.Sequence, rdc)
+		}
+	}
+
+	// Do one final check here.
+	o.mu.Lock()
+	defer o.mu.Unlock()
+
+	// If we are here, and this should be rare, we still are off with our ack floor.
+	// We will set it explcitly to 1 behind our current lowest in pending, or if
```
Comment (neilalexander) [inline server/consumer.go:3282]: Explicitly

**Review by neilalexander — APPROVED** [review]
LGTM

**Review by wallyqs — APPROVED** [review]
LGTM

File: server/consumer.go (hunk 1 above)
Comment (derekcollison) [reply server/consumer.go:3282]: Thanks will fix.

Comment (derekcollison) [reply server/consumer.go]: Thanks will fix.

**Pull request merged**